### PR TITLE
Adding rake:publish

### PR DESCRIPTION
**Pull request by jjasghar**
This will allow for a release of what Tyler Ball want's as a standard
release process.

- Bumps the minor version
- runs github_changelog_generator
- commits the things to github
- _will_ push to rubygems...after everyone is cool with this PR.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+  sh "git commit -am 'Preparing for #{future_version} release' && git push"
+
+  # Runs git tag to take the release (git tag -a "Releasing #{future_version}" && git push --tags)
+  sh "git tag -a 'Releasing #{future_version}' && git push --tags"
+
+  # need to add rake release here :)
+end
+
+desc 'Does a patch bump to the gem'
+task :minor do
```
Comment (adamleff) [inline Rakefile]: Probably should be `:patch`

File: Rakefile [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
 require 'bundler/gem_tasks'
+require 'bump'
+require 'github_changelog_generator'
 
 task :spec do
   require File.expand_path('spec/run')
 end
+
+desc 'Sanity Checks and releases gem to rubygems'
+task :sanity do
```
Comment (adamleff) [inline Rakefile]: This block doesn't read like a "sanity check" block - it's actually making changes. Call it `:release` or something else that more clearly states what it's doing.

File: Rakefile (hunk 1 above)
Comment (jjasghar) [inline Rakefile]: touche`

File: Gemfile [hunk 3]
```diff
@@ -8,4 +8,6 @@ group :development do
   gem 'guard'
   gem 'guard-rspec'
   gem 'rb-readline'
+  gem 'bump'
```
Comment (adamleff) [inline Gemfile]: Shouldn't all these gems in the :development group really exist in the specfile as a development_dependency?

File: Rakefile (hunk 2 above)
Comment (jjasghar) [inline Rakefile]: there already is a `rake release` wasn't sure what to name it. `:sanity` seemed...sane? maybe `rake push`?

Comment (adamleff) [inline Rakefile]: `publish` or `push` makes more sense to me. Sanity checks should not actually modify anything - they should be a gate.

File: Gemfile (hunk 3 above)
Comment (jjasghar) [inline Gemfile]: Good call, I'm moving them now.

File: Rakefile [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+  sh 'github_changelog_generator'
+
+  # Commits those as a change to master (git commit -am "Preparing the #{future_version} release" && git push)
+  future_version = Bump::Bump.current
+  sh "git commit -am 'Preparing for #{future_version} release' && git push"
+
+  # Runs git tag to take the release (git tag -a "Releasing #{future_version}" && git push --tags)
+  sh "git tag -a 'Releasing #{future_version}' && git push --tags"
+
+  # need to add rake release here :)
```
Comment (tyler-ball) [inline Rakefile]: `rake release` will do the tagging for you, so when that is added you won't need the above line

File: chef-provisioning-fog.gemspec [hunk 5]
```diff
@@ -12,12 +12,18 @@ Gem::Specification.new do |s|
   s.email = ['jkeiser@getchef.com', 'hh@vulk.co', 't@vulk.co', 'w@vulk.co']
   s.homepage = 'https://github.com/opscode/chef-provisioning-fog'
 
+  s.add_dependnecy 'chef'
```
Comment (tyler-ball) [inline chef-provisioning-fog.gemspec]: `chef-provisioning` should pull this in automatically - can you remove this and run `bundle update` to check if chef is still being pulled in?

**Comment by tyler-ball** [comment]
Looking good so far!  Can I help with anything?

**Pull request closed without merging**